source/val/validate_constants.cpp

```diff
@@ -282,6 +288,142 @@ spv_result_t ValidateConstantComposite(ValidationState_t& _,
                << _.getIdName(result_type->id()) << "s component type.";
       }
     } break;
+    case spv::Op::OpTypeTensorARM: {
+      auto inst_element_type =
+          _.FindDef(result_type->GetOperandAs<uint32_t>(1));
+      if (!inst_element_type) {
+        return _.diag(SPV_ERROR_INVALID_ID, result_type)
+               << "Element type is not defined.";
+      }
+      const auto inst_rank = _.FindDef(result_type->GetOperandAs<uint32_t>(2));
+      if (!inst_rank) {
+        return _.diag(SPV_ERROR_INVALID_ID, result_type)
+               << "Rank is not defined.";
+      }
+      const auto inst_shape = _.FindDef(result_type->GetOperandAs<uint32_t>(3));
+      if (!inst_shape) {
+        return _.diag(SPV_ERROR_INVALID_ID, result_type)
+               << "Shape is not defined.";
+      }
+
+      uint64_t rank = 0;
+      if (!_.EvalConstantValUint64(inst_rank->id(), &rank)) {
+        return _.diag(SPV_ERROR_INVALID_ID, result_type)
```

Comment:
The extension allows Rank and Shape to be spec constants.
But the _.EvalConstantValUint64 method returns false for spec constants.

So validation is more strict than the extension.

Reply:
Good catch! This was changed upstream a while ago but we have a revert internally _and_ I don't have unit tests for these cases.